subprojects/docs/src/docs/userguide/dependency_management_attribute_based_matching.adoc

```diff
@@ -0,0 +1,206 @@
+Users who simply want to understand what configuration attributes are will also find support here.
+
+[[sec:abm_configuration_kinds]]
+== Different kinds of configurations
+
+Historically, <<managing_dependency_configurations.adoc#sub:what_is_a_configuration,configurations>>  have been at the root of dependency resolution in Gradle.
+In the end, what we want to make a diffence is between a _consumer_ and a _producer_. For this purpose, configurations are used for at least 3 different aspects:
+
+1. to declare dependencies
+2. as a _consumer_, to resolve a set of dependencies to files
+3. as a _producer_, to expose artifacts for consumption by other projects
+
+For example, if I want to express that my application `app` _depends on_ library `lib`, we need _at least_ one configuration:
+
+[source,groovy]
+----
+configurations {
+   // declare a "configuration" named "someConfiguration"
+   someConfiguration
+}
+dependencies {
+   // add a project dependency to the "someConfiguration" configuration
+   someConfiguration project(":lib")
+}
+----
+
+Configurations can extend other configuration, in order to inherit their dependencies.
+However, the code above doesn't tell anything about the _consumer_.
+In particular, it doesn't tell what is the _use_ of the configuration.
+Let's say that `lib` is a Java library: it can expose different things, such as its API, implementation or test fixtures.
+If we want to resolve the dependencies of `app`, we need to know what kind of task we're performing (compiling against the API of `lib`, executing the application, compiling tests, ...).
+For this purpose, you'll often find companion configurations, which are meant to unambiguously declare the usage:
+
+[source,groovy]
+----
+configurations {
+    // declare a configuration that is going to resolve the compile classpath of the application
+    compileClasspath.extendsFrom(someConfiguration)
+
+    // declare a configuration that is going to resolve the runtime classpath of the application
+    runtimeClasspath.extendsFrom(someConfiguration)
+}
+----
+
+At this stage, we have 3 different configurations, which already have different goals:
+
+- `someConfiguration` declares the dependencies of my application. It's just a bucket where we declare a list of dependencies.
+- `compileClasspath` and `runtimeClasspath` are configurations _meant to be resolved_: when resolved they should contain respectively the compile classpath, and the runtime classpath of the application.
+
+This is actually represented on the `Configuration` type by the `canBeResolved` flag. A configuration that _can be resolved_ is a configuration which is going to be, at some point, resolved.
+That is to say we're going to compute a dependency graph, resolve the components in the graph, and eventually get artifacts.
+A configuration which has `canBeResolved` set to `false` is not meant to be resolved.
+Such a configuration is there _only to declare dependencies_.
```

Comment:
> only to declare dependencies

why is this useful? Is it just so other "resolvable" configurations can extend it, kinda like an abstract class?